### DISCUSSION TITLE
Feature Request: props.defaultValue as string element

### DISCUSSION THREAD

**Discussion by mkikets99** [JedWatson/react-select#4453] [Q&A]
# Are you asking a question?
This is feature request

# Are you making a feature request?
Provide as much information as possible about your requested feature. Here are a few questions you may consider answering:

***Note:*** 
```javascript
typeof this.state.edata.relates === 'string' //true
```

*   What's your use case? (Tell us about your application and what problem you're trying to solve.)
    - This plugin is very helpful but it has very hard implementation of default value in grouped select ( something like this is needed:
    ```jsx
    <Select defaultValue={_.find(_.flatten(_.map(this.state.data,(el)=>el.options)),{value:this.state.edata.relates})} ... />
    ```      
*   What interface do you have in mind? (What new properties or methods do you think might be helpful?)
    *   Using it like this for grouped select would be much easier
    ```jsx
    <Select defaultValue={this.state.edata.relates} ... />
    ```   
*   Can you point to similar functionality with any existing libraries or components? (Working demos can be helpful.)
    (I dunno.) Maybe, something like on MDB premium:
    https://mdbootstrap.com/docs/react/forms/select/

**Comment by ebonow** [comment]
Greetings @mkikets99, 

`value` and `defaultValue` were both originally implemented in v1 as string types with support for simpleValues, but this was later removed in v2.

I believe your ask here is different in that you wish react-select to traverse the options and automatically select the option with the matching value. This introduces a lot of complexity with the underlying Types as the primary intention of a defaultValue is to share the same value Type as an Option and not to exist as a parameter to some predicate function.

There has been some talk about how confusing "value" is. While major versions 3, 4, and soon to be 5 have not introduced a lot of breaking changes, version 6 will likely be much different. One of the discussions is to replace what is now known as `value` with `selected`. I think this would behave much closer to a native select element where value is understood to represent the value of a selected option.

However for time being, value and defaultValue are meant to be an `option` object. Since this was addressed as an issue I will transfer this to the Discussions area with other feature requests to have more transparency and support for the long term future direction of react-select.
